x-pack/plugins/cases/public/application.tsx

```diff
@@ -37,17 +39,23 @@ export const renderApp = (deps: RenderAppProps) => {
 interface CasesAppWithContextProps {
   externalReferenceAttachmentTypeRegistry: ExternalReferenceAttachmentTypeRegistry;
   persistableStateAttachmentTypeRegistry: PersistableStateAttachmentTypeRegistry;
+  filesPlugin: FilesStart;
```

Comment:
Maybe instead of passing the whole files plugin we can pass a function that gives you the files client. For example: `getFilesClient: (fileKind) => ScopedFilesClient` where `getFilesClient = filesPlugin.filesClientFactory.asScoped`